marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java

```diff
@@ -246,13 +250,23 @@ private void addJobIdToEndpointConstants() {
         if (endpointParams != null && endpointParams.has("endpointConstants")) {
             JsonNode endpointConstants = endpointParams.get("endpointConstants");
             if (endpointConstants instanceof ObjectNode) {
-                ((ObjectNode)endpointConstants).put("jobId", this.jobId);
+                ((ObjectNode) endpointConstants).put("jobId", this.jobId);
             }
         }
     }
 
     private void finalizeJob(String status) {
+        String apiPath = "/marklogic-data-hub-spark-connector/finalizeJob.api";
```

Comment:
I think we need to initialize these modules before we create the job document. That ensures we fail fast if there's any problem - i.e. we don't want to load a bunch of data and then have an error if we can't write these modules for some reason.